rust/xaynet-analytics/src/data_combination/data_points/was_active_each_past_period.rs

```diff
@@ -52,9 +52,153 @@ impl CalculateDataPoints for CalcWasActiveEachPastPeriod {
 
     fn calculate(&self) -> Vec<u32> {
         let timestamps_by_period_threshold = self.group_timestamps_by_period_threshold();
+        // since we are travelling 'back in time' we need to reverse the order of the values of the BTreeMap
         timestamps_by_period_threshold
             .values()
+            .rev()
             .map(|timestamps| !timestamps.is_empty() as u32)
             .collect::<Vec<u32>>()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use chrono::{DateTime, Duration, Utc};
+
+    use crate::{
+        data_combination::data_points::{
+            data_point::{CalculateDataPoints, DataPointMetadata, Period, PeriodUnit},
+            was_active_each_past_period::CalcWasActiveEachPastPeriod,
```

Comment:
you can replace some imports with `use super::*` which imports everything that is defined in `mod was_active_each_past_period;`

Reply:
i wasn't sure what's best practice with Rust, whether to be more explicit or more practical. i'll use more `use super::*` then! thanks

Reply:
For the `tests` mod, it is fine to use` super :: * `because you usually want to test what is in the parent module

Reply:
agree, in general more explicit is always better and i personally tend to only import explicitly, even for things like `prelude`s. of course it is more convenient to import everything implicitly, but it is way easier to read for other if imported explicitly.
that said, the only exception i make is for `super::*` in tests, because the tests refer directly to the code and are usually even written in the same file, which leaves very little room for ambiguity and hidden things.

Reply:
sounds good, totally agree